app/views/work_packages/calendars/index.html.erb

```diff
@@ -28,62 +28,55 @@ See doc/COPYRIGHT.rdoc for more details.
-      </li>
-    </ul>
+    <%= hidden_field_tag('project_id', @project.to_param) if @project %>
+    <%= hidden_field_tag('set_filter', 1) %>
+    <fieldset id="filters" class="form--fieldset -collapsible collapsed">
+      <legend class="form--fieldset-legend" onclick="toggleFieldset(this);">
+        <a href="javascript:"><%= l(:label_filter_plural) %></a></legend>
+      <div class="" style="display: none;">
+        <%= render partial: 'queries/filters', locals: { query: @query } %>
+      </div>
+    </fieldset>
+    <fieldset class="simple-filters--container">
+      <legend><%= l('timeframe.show') %></legend>
+      <ul class="simple-filters--filters">
+        <li class="simple-filters--filter">
+          <%= label_tag('month', l(:label_month), class: 'simple-filters--filter-name') %>
+          <%= select_month(@month, { prefix: "month", discard_type: true }, class: 'simple-filters--filter-value') %>
+        </li>
+        <li class="simple-filters--filter">
+          <%= label_tag('year', l(:label_year), class: 'simple-filters--filter-name') %>
+          <%= select_year(@year, { prefix: "year", discard_type: true }, class: 'simple-filters--filter-value') %>
+        </li>
+        <li class="simple-filters--controls">
+          <%= submit_tag l(:button_apply), class: 'button -highlight -small' %>
+        </li>
+      </ul>
 
-  </fieldset>
+    </fieldset>
 <% end %>
 <%= error_messages_for 'query' %>
 <% if @query.valid? %>
-  <p style="float:right;">
-    <%= link_to_previous_month(@year, @month, project: @project) %> | <%= link_to_next_month(@year, @month, project: @project) %>
-  </p>
-  <%= render partial: 'common/calendar', locals: {calendar: @calendar} %>
-  <p class="legend cal">
-    <span class="starting icon icon-arrow-right4"></span>
-    <%= l(:text_tip_work_package_begin_day) %>
-  </p>
-  <p class="legend cal">
-    <span class="ending icon icon-arrow-right4"></span>
-    <%= l(:text_tip_work_package_end_day) %>
-  </p>
-  <p class="legend cal">
-    <span class="icon icon-milestone diamond"></span>
-    <%= l(:text_tip_work_package_begin_end_day) %>
-  </p>
+    <p style="float:right;">
+      <%= link_to_previous_month(@year, @month, project: @project) %>
+      | <%= link_to_next_month(@year, @month, project: @project) %>
+    </p>
+    <%= render partial: 'common/calendar', locals: { calendar: @calendar } %>
+    <p class="legend cal">
+      <span class="starting icon icon-arrow-right4"></span>
+      <%= l(:text_tip_work_package_begin_day) %>
+    </p>
+    <p class="legend cal">
+      <span class="ending icon icon-arrow-right4"></span>
+      <%= l(:text_tip_work_package_end_day) %>
+    </p>
+    <p class="legend cal">
+      <span class="icon icon-milestone diamond"></span>
+      <%= l(:text_tip_work_package_begin_end_day) %>
+    </p>
 <% end %>
 <% content_for :sidebar do %>
-  <%= render partial: 'work_packages/sidebar' %>
+    <%= render partial: 'work_packages/sidebar' %>
```

Comment:
Fixed in https://github.com/opf/openproject/pull/4323/commits/5c48cdd7c243165a28315a41e23534e18948185f